src/challenges/challengeOne/4.Form/Form.test.tsx

```diff
@@ -1,8 +1,81 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Form } from "./Form";
-
-// describe("Form", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Form, FormProps, FormField } from "./Form";
+
+describe("Form", () => {
+  const fields: FormField[] = [
+    {
+      label: "Name",
+      type: "text",
+      name: "name",
+      placeholder: "Enter your name",
+      value: "",
+      id: "name",
+    },
+    {
+      label: "Email",
+      type: "email",
+      name: "email",
+      placeholder: "Enter your email",
+      value: "",
+      id: "email",
+    },
+  ];
+
+  const handleSubmit = jest.fn();
+
+  const renderComponent = (additionalProps?: Partial<FormProps>) => {
+    const defaultProps: FormProps = {
+      fields,
+      onSubmit: handleSubmit,
+    };
+    render(<Form {...defaultProps} {...additionalProps} />);
+  };
+
+  it("renders the form fields correctly", () => {
+    renderComponent();
+
+    fields.forEach((field) => {
+      const label = screen.getByText(field.label);
+      const input = screen.getByLabelText(field.label);
+
+      expect(label).toBeInTheDocument();
+      expect(input).toBeInTheDocument();
+      expect(input).toHaveAttribute("type", field.type);
+      expect(input).toHaveAttribute("name", field.name);
+      expect(input).toHaveAttribute("placeholder", field.placeholder);
+      expect(input).toHaveValue(field.value);
+    });
+  });
+
+  it("updates the form values correctly", () => {
+    renderComponent();
+
+    const nameInput = screen.getByLabelText("Name");
+    const emailInput = screen.getByLabelText("Email");
+
+    fireEvent.change(nameInput, { target: { value: "John" } });
```

Comment:
I left `userEvent` as an imported package as it is superior to `fireEvent`. `userEvent` minics the user interactions with the UI, which is what we're always trying to test. `fireEvent` directly interacts with the dom, which is not what we want to test.